tool/site/server/src/routes/auth/login.ts

```diff
@@ -20,6 +19,20 @@ interface UnattendedTokenResponse {
   token_type: "Bearer"
 }
 
+function getRedirectUri(authorizeUrl: string, clientId: string, callbackUri: string, scopes?: Array<string>) {
+  const queryParams: Record<string, string> = {
+    "client_id": clientId,
+    "redirect_uri": callbackUri,
+    "response_type": "code",
+    "state": uuid.v4()
+  }
+  if (scopes) {
```

Comment:
I generally don't like having optional function parameters if I can avoid it, we could make `scopes` required and then change this to be `if (scopes.length > 0)`, I understand if others disagree though